Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
         /* When fut is subclass of Future, finalizer is called from
          * subtype_dealloc.
          */
+         PyObject_GC_Track(self);
```

Comment:
Yes, please.  asyncio speedup module is introduced from Python 3.6.
It make easy to backport.